server/schema.go

```diff
@@ -2,26 +2,147 @@ package server
 
 import (
 	"context"
+	"fmt"
+
+	"github.com/batchcorp/plumber-schemas/build/go/protos/encoding"
+
+	"github.com/pkg/errors"
+
+	"github.com/batchcorp/plumber/server/types"
+
+	uuid "github.com/satori/go.uuid"
 
 	"github.com/batchcorp/plumber-schemas/build/go/protos"
+	"github.com/batchcorp/plumber-schemas/build/go/protos/common"
 )
 
 func (p *PlumberServer) GetAllSchemas(_ context.Context, req *protos.GetAllSchemasRequest) (*protos.GetAllSchemasResponse, error) {
-	return nil, nil
+	if err := p.validateRequest(req.Auth); err != nil {
+		return nil, CustomError(common.Code_UNAUTHENTICATED, fmt.Sprintf("invalid auth: %s", err))
+	}
+
+	schemas := make([]*protos.Schema, 0)
+
+	for _, v := range p.PersistentConfig.Schemas {
+		schemas = append(schemas, v.Schema)
+	}
+
+	return &protos.GetAllSchemasResponse{
+		Schema: schemas,
+	}, nil
 }
 
 func (p *PlumberServer) GetSchema(_ context.Context, req *protos.GetSchemaRequest) (*protos.GetSchemaResponse, error) {
-	return nil, nil
+	if err := p.validateRequest(req.Auth); err != nil {
+		return nil, CustomError(common.Code_UNAUTHENTICATED, fmt.Sprintf("invalid auth: %s", err))
+	}
+
+	schema := p.getSchema(req.Id)
+	if schema == nil {
+		return nil, CustomError(common.Code_NOT_FOUND, "schema does not exist")
+	}
+
+	return &protos.GetSchemaResponse{
+		Schema: schema.Schema,
+	}, nil
 }
 
 func (p *PlumberServer) ImportGithub(ctx context.Context, req *protos.ImportGithubRequest) (*protos.ImportGithubResponse, error) {
-	return nil, nil
+
+	schema, err := p.importGithub(ctx, req)
+	if err != nil {
+		return nil, CustomError(common.Code_FAILED_PRECONDITION, err.Error())
+	}
+
+	return &protos.ImportGithubResponse{
+		Status: &common.Status{
+			Code:      common.Code_OK,
+			Message:   "schema imported successfully",
+			RequestId: uuid.NewV4().String(),
+		},
+		Id: schema.Schema.Id,
+	}, nil
+}
+
+// importGithub imports a github repo as a schema
+// TODO: types other than protobuf
+func (p *PlumberServer) importGithub(ctx context.Context, req *protos.ImportGithubRequest) (*types.Schema, error) {
+	zipfile, err := p.GithubService.GetRepoArchive(ctx, p.PersistentConfig.GitHubToken, req.GithubUrl)
+	if err != nil {
+		return nil, err
+	}
+
+	var schema *types.Schema
```

Comment:
Let's clean this up so that it's ready for other formats. Change the if into a switch {} + move the proto stuff into their own method.

Reply:
Fixed!